storage/innobase/srv/srv0srv.cc

```diff
@@ -1304,12 +1304,31 @@ static tpool::waitable_task purge_truncation_task
 static tpool::timer *purge_coordinator_timer;
 
 /** Wake up the purge threads if there is work to do. */
-void srv_wake_purge_thread_if_not_active()
+void purge_sys_t::wake_if_not_active()
 {
-  if (purge_sys.enabled() && !purge_sys.paused() &&
-      (srv_undo_log_truncate || trx_sys.history_exists()) &&
-      ++purge_state.m_running == 1)
-    srv_thread_pool->submit_task(&purge_coordinator_task);
+  if (enabled() && !paused() && !purge_state.m_running)
+  {
+    if (srv_undo_log_truncate);
+    else if (!trx_sys.history_exists())
+      return;
+    else
+    {
+      purge_coordinator_timer->disarm();
```

Comment:
Can we move this complicated logic/locking into purge_coordinator_task? basically doing as before? A false wakeup would not happen very often with the old logic either?

Reply:
Done in ee1ff06819b624446b50cb3c14019b7c15469498